app/src/main/java/com/nervousfish/nervousfish/service_locator/ServiceLocatorNoNetwork.java

```diff
@@ -0,0 +1,119 @@
+package com.nervousfish.nervousfish.service_locator;
+
+import com.nervousfish.nervousfish.ConstantKeywords;
+import com.nervousfish.nervousfish.modules.constants.Constants;
+import com.nervousfish.nervousfish.modules.constants.IConstants;
+import com.nervousfish.nervousfish.modules.cryptography.EncryptorAdapter;
+import com.nervousfish.nervousfish.modules.cryptography.IEncryptor;
+import com.nervousfish.nervousfish.modules.cryptography.IKeyGenerator;
+import com.nervousfish.nervousfish.modules.cryptography.KeyGeneratorAdapter;
+import com.nervousfish.nervousfish.modules.database.GsonDatabaseAdapter;
+import com.nervousfish.nervousfish.modules.database.IDatabase;
+import com.nervousfish.nervousfish.modules.filesystem.AndroidFileSystemAdapter;
+import com.nervousfish.nervousfish.modules.filesystem.IFileSystem;
+import com.nervousfish.nervousfish.modules.pairing.AndroidBluetoothHandler;
+import com.nervousfish.nervousfish.modules.pairing.DummyBluetoothHandler;
+import com.nervousfish.nervousfish.modules.pairing.DummyNFCHandler;
+import com.nervousfish.nervousfish.modules.pairing.DummyQRHandler;
+import com.nervousfish.nervousfish.modules.pairing.IBluetoothHandler;
+import com.nervousfish.nervousfish.modules.pairing.INfcHandler;
+import com.nervousfish.nervousfish.modules.pairing.IQRHandler;
+
+import org.greenrobot.eventbus.EventBus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InvalidObjectException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+/**
+ * Manages all modules and provides access to them.
+ * It disables Bluetooth and NFC because that cannot be used on an emulator
+ */
+public class ServiceLocatorNoNetwork extends ServiceLocator implements IServiceLocator {
```

Comment:
Without this service locator several Cucumber tests cannot run. For example, after the user has filled out the correct pattern in the grid activity, the app returns to MainActivity. MainActivity starts the Bluetooth service to listen for incoming pairing requests and that crashes of course on an emulator. I decided to call it "NoNetwork" to denote that this class has no network functionality in a fairly broad sense. Not only WiFi is over a network, but a network is a group of interconnected things, so also Bluetooth, NFC, an EventBus and even QR when you use it to communicate between devices. If you have a better name, please let me know ;-)
![capture](https://cloud.githubusercontent.com/assets/15014738/26553427/e56910ce-448b-11e7-8d3a-c963c9da7740.JPG)

I also agree that test code shouldn't affect production code, but when we have to choose to either leave out all tests that start MainActivity or modify ServiceLocator in such a way that we do can test everything, I think that testing that stuff actually works is more important that code style.